src/lib.rs

```diff
@@ -151,8 +132,8 @@ pub fn metadata_deps(manifest_path_arg: Option<&str>, deps: bool) -> Result<Meta
     if let Some(mani) = manifest_path_arg {
         cmd.arg(mani);
     }
-    let output = cmd.output()?;
-    let stdout = from_utf8(&output.stdout)?;
-    let meta: Metadata = serde_json::from_str(stdout)?;
+    let output = cmd.output().chain_err(|| "Failed to execute `cargo metadata`")?;
```

Comment:
hmm... I actually like that we had the real error in here. Can you reintroduce the variants via error_chain?